### PR TITLE
More auto tests

### PR DESCRIPTION
**Pull request by nailed1**
# Расширение автотестов

### Инфра
- Добавлена конфигурация pytest в `pyproject.toml` (`pythonpath = ["."]`, `testpaths = ["Tests"]`) - тесты теперь запускаются одной командой без ручной настройки `PYTHONPATH`
- Добавлены `Tests/__init__.py` и `Src/__init__.py` для корректного разрешения пакетов
- Добавлен базовый класс `DPGUnitTestWithReset` - расширение `DPGUnitTest`, которое сбрасывает кэш `ThemeManager` между тестовыми классами. Это устраняет падения, когда несколько классов пересоздают DPG-контекст и старые ID тем становятся невалидны

### Фиксы
- Исправлен путь импорта `ThemeManager`: `Src.Themes` -> `Src.Managers.theme_manager`
- Все тестовые классы, создающие DPG-контекст, переведены на `DPGUnitTestWithReset`
- `test_backfield` адаптирован под новое API Backfield: колбэк теперь не передаётся в конструктор, а привязывается через `Backfield.bind(instance, callback)`
- `test_compilation` исправлен: обновлены ключи `node_list` (`"Preprocessing Utils"` -> `"Processing Utils"`), индексы нод и регистр значений enum (`softmax` -> `SOFTMAX`, `binary_crossentropy` -> `BINARY_CROSSENTROPY`)

### Новые тесты

| Файл | Новых тестов | Что покрыто |
|------|:------------:|-------------|
| `test_nodes.py` | 16 | `LayerNode`, `InputLayerNode`, `CompileNode`, `FitNode`, `PredictNode`, `ShapeNode`, `MetricNode`, `UtilsNode` - без DPG, только keras/numpy |
| `test_parameter.py` | +8 | `AFloat`, `ABoolean`, `AString`, `AttrType.OUTPUT`/`STATIC`, некорректный тип, отсутствие default, несовместимый родитель |
| `test_annotations.py` | +4 | `Annotation.check_kwargs` (пустой ввод, нет совпадений), пустая строка в `AString`, отрицательное значение в `AInteger` |
| `test_backfield.py` | +2 | Сохранение значения после присвоения, колбэк получает правильное значение |
| `test_themes.py` | +2 | Повторный `apply_theme` заменяет предыдущую тему, `get_theme` возвращает валидный DPG-элемент |
| `test_node_editor.py` | +2 | Тег `"node_editor"` существует в DPG, `drop_callback` возвращает валидный ID ноды |
| `test_node_builder.py` | +1 | Нода получает метку из `NodeAnnotation` |
| `test_compilation.py` | +1 | `compile_graph` возвращает `set` |

## Запуск тестов

```bash
# Все тесты (без интеграционного test_compilation)
pytest Tests/ --ignore=Tests/test_compilation.py

# Только чистые unit-тесты (без DPG)
pytest Tests/test_nodes.py Tests/test_backfield.py

# С подробным выводом
pytest Tests/ --ignore=Tests/test_compilation.py -v

# Полный прогон включая компиляцию графа (потребует времени)
pytest Tests/
```

## Coverage

**60 тестов пройдено / 60 собрано**  
Coverage report

| Name | Stmts | Miss | Cover |
|------|------:|-----:|------:|
| `Src\Config\parameter.py` | 35 | 0 | 100% |
| `Src\Config\node_annotation.py` | 31 | 0 | 100% |
| `Src\Nodes\compile_node.py` | 15 | 0 | 100% |
| `Src\Nodes\fit_node.py` | 29 | 0 | 100% |
| `Src\Nodes\input_layer_node.py` | 10 | 0 | 100% |
| `Src\Nodes\layer_node.py` | 20 | 0 | 100% |
| `Src\Nodes\predict_node.py` | 9 | 0 | 100% |
| `Src\Nodes\data_node.py` | 4 | 0 | 100% |
| `Src\Utils\backfield.py` | 17 | 0 | 100% |
| `Src\Config\Annotations\annotation.py` | 19 | 0 | 100% |
| `Src\Managers\theme_manager.py` | 58 | 1 | 98% |
| `Src\Config\Annotations\anot_enum.py` | 24 | 1 | 96% |
| `Src\Config\Annotations\anot_file.py` | 26 | 1 | 96% |
| `Src\Config\Annotations\anot_bool.py` | 18 | 1 | 94% |
| `Src\Config\Annotations\anot_float.py` | 18 | 1 | 94% |
| `Src\Config\Annotations\anot_integer.py` | 18 | 1 | 94% |
| `Src\Config\Annotations\anot_string.py` | 18 | 1 | 94% |
| `Src\node_editor.py` | 117 | 10 | 91% |
| `Src\node_builder.py` | 84 | 11 | 87% |
| `Src\Nodes\utils_node.py` | 14 | 2 | 86% |
| `Src\Nodes\abstract_node.py` | 90 | 21 | 77% |
| `Src\Nodes\shape_node.py` | 31 | 8 | 74% |
| `Src\Nodes\metric_node.py` | 18 | 5 | 72% |
| `Src\Nodes\dataset_node.py` | 42 | 16 | 62% |
| `Src\Managers\event_manager.py` | 26 | 15 | 42% |
| `Src\Utils\factory_method.py` | 22 | 13 | 41% |
| `Src\Managers\callbacks.py` | 108 | 85 | 21% |
| **TOTAL** | **1416** | **203** | **86%** |

## Summary by CodeRabbit

* **Tests**
  * Expanded unit test coverage for annotations, parameters, node builders, and compilation features
  * Added validation tests for data processing pipelines, layer operations, inference, and metrics
  * Improved test initialization with optimized theme management to prevent cache persistence across test suites
  * Enhanced assertion coverage for model training, prediction, and JSON serialization

* **Chores**
  * Updated test configuration for improved test discovery and execution

### CODE REVIEW

File: Tests/test_node_builder.py [hunk 1]
```diff
@@ -6,20 +6,23 @@
...
     '''
     Проверка объектов аннотации и их работоспособности
     '''
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
         dpg.create_viewport(title='Custom Title')
+        # Перезагружаем темы явно — после пересоздания контекста
+        # старые ID тем из предыдущего тестового класса становятся невалидны
+        from Src.Managers.theme_manager import ThemeManager
```
Comment (jezvgg) [inline Tests/test_node_builder.py]: А зачем импортируем тут?

**Review by jezvgg — APPROVED** [review]
Отличная работа.

**Comment by jezvgg** [comment]
@coderabbitai help
